### PR TITLE
Add DNS API support for Nexcess, Thermo.io, and Futurehosting

### PR DESCRIPTION
**Pull request by laszlof**
This adds DNS support for [Nexcess](https://www.nexcess.net), [Futurehosting](https://www.futurehosting.com), and [Thermo.io](https://www.thermo.io)

All 3 use the same API backend (different endpoints), so the code is mostly identical other than the names of things.

### CODE REVIEW

**Comment by Neilpang** [comment]
please check this guide https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide

**Comment by laszlof** [comment]
@Neilpang Thanks. I have read through the guide. Is there something specific I am missing from it?

**Comment by laszlof** [comment]
@Neilpang is there anything additional that is needed on this PR?

**Comment by laszlof** [comment]
Bumping this back up. Anything needed on my end @Neilpang?

**Comment by laszlof** [comment]
Merge Conflicts Resolved.

**Comment by laszlof** [comment]
Conflicts Resolved. (Again).

File: dnsapi/dns_fh.sh [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    return 1
+  fi
+  _debug _zone_id "${_zone_id}"
+  _debug _sub_domain "${_sub_domain}"
+  _debug _domain "${_domain}"
+
+  _post_data="{\"zone_id\": \"${_zone_id}\", \"type\": \"TXT\", \"host\": \"${host}\", \"target\": \"${txtvalue}\", \"ttl\": \"300\"}"
+
+  if _rest POST "dns-record" "${_post_data}" && [ -n "${response}" ]; then
+    _record_id=$(printf "%s\n" "${response}" | _egrep_o "\"record_id\":\s*[0-9]+" | cut -d : -f 2 | tr -d " " | _head_n 1)
```
Comment (Neilpang) [inline dnsapi/dns_fh.sh]: do not use `'\s'`,  it's not crossing platforms.
please just use plain space char `' '` instead.

File: dnsapi/dns_fh.sh [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    return 0
+  fi
+
+  return 1
+}
+
+_check_fh_api_token() {
+  if [ -z "${FH_API_TOKEN}" ]; then
+    FH_API_TOKEN="${FH_API_TOKEN:-$(_readaccountconf FH_API_TOKEN)}"
```
Comment (Neilpang) [inline dnsapi/dns_fh.sh]: please use `_readaccountconf_mutable()` and `_saveaccountconf_mutable()` function instead.
see the `dns_cf.sh`

**Comment by Neilpang** [comment]
https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide

please make sure you can handle all the domains:
```
_acme-challenge.www.example.com
_acme-challenge.example.com
_acme-challenge.example.co.uk
_acme-challenge.www.example.co.uk
_acme-challenge.sub1.sub2.www.example.co.uk
```

And support v2 wildcard:
https://github.com/Neilpang/acme.sh/issues/1261

**Comment by laszlof** [comment]
All issues have been corrected.

Also confirmed that it works with all example, including v2 wildcard.

**Comment by Neilpang** [comment]
Sorry,  I just noticed that you said these 3 apis are identical.  
I wonder if we can use only script to handle all the 3 ?

```sh
export   XXX_AccessToken=xxxxxxx
export  XXX_Endpoint="https://my.futurehosting.com/"

acme.sh --issue  -d example.com  --dns  dns_xxx.sh 
```

We can save `XXX_AccessToken` and `XXX_Endpoint`

what do you think ?

**Comment by laszlof** [comment]
@Neilpang  Thats actually a good idea. I'll work out the details and update the PR

**Comment by laszlof** [comment]
@Neilpang This has been tested and should be good to go.

**Comment by laszlof** [comment]
Done. https://github.com/Neilpang/acme.sh/pull/2082

**Pull request closed without merging**